### PR TITLE
Set cursor to left_ptr when mouse is over a notification on wayland

### PR DESCRIPTION
**Pull request by GorrillaRibs**
As in mako here: https://github.com/emersion/mako/commit/85a3e38e9334cdd31dd43802b6850255986a72f2

Currently (on wayland) the cursor remains what it was when the mouse enters the notification window (i.e. from a terminal or a text entry box it'll be a vertical bar), this fixes that.

Code is taken from [mako](https://github.com/emersion/mako) (which used to have the same issue) and adapted a little to fit dunst.

### CODE REVIEW

File: config.mk [hunk 1]
```diff
@@ -56,4 +56,5 @@ pkg_config_packs += libnotify
 
 ifneq (0,${WAYLAND})
 pkg_config_packs += wayland-client
+pkg_config_packs +=wayland-cursor
```
Comment (fwSmit) [inline config.mk:59]:
```suggestion
pkg_config_packs += wayland-cursor
```

File: src/wayland/wl.c [hunk 2]
```diff
@@ -70,17 +71,20 @@ struct wl_ctx {
         } pointer;
 
 	struct {
-		struct wl_touch *wl_touch;
-		struct {
-			int32_t x, y;
-		} pts[MAX_TOUCHPOINTS];
-	} touch;
+                struct wl_touch *wl_touch;
```
Comment (fwSmit) [inline src/wayland/wl.c:74]: Please change the tabs to spaces. In the rest of the code as well

File: src/wayland/wl.c [hunk 3]
```diff
@@ -579,6 +590,40 @@ bool wl_init() {
                 LOG_W("compositor doesn't support zwlr_foreign_toplevel_v1. Fullscreen detection won't work");
         }
 
+	// Set up the cursor. It needs a wl_surface with the cursor loaded into it.
+	// If one of these fail, mako will work fine without the cursor being able to change.
```
Comment (fwSmit) [inline src/wayland/wl.c:594]:
```suggestion
	// If one of these fail, dunst will work fine without the cursor being able to change.
```

File: src/wayland/wl.h [hunk 4]
```diff
@@ -28,5 +28,6 @@ bool wl_have_fullscreen_window(void);
 // should be multiplied by this value, but don't use it to multiply other
 // values. All sizes should be in unscaled units.
 double wl_get_scale(void);
+
```
Comment (fwSmit) [inline src/wayland/wl.h:31]: Could you remove this unrelated change?

**Review by fwsmit — CHANGES_REQUESTED** [review]
Thanks for porting this patch. I've got a few style comments, but other than that it looks good

File: src/wayland/wl.c (hunk 2 above)
Comment (GorrillaRibs) [reply src/wayland/wl.c:74]: Done (I think - tried to do it automatically w/ nvim and it looks ok, but probably worth double checking)

File: src/wayland/wl.h (hunk 4 above)
Comment (GorrillaRibs) [reply src/wayland/wl.h]: Done!

**Comment by GorrillaRibs** [comment]
I think I've fixed everything brought up, but I haven't really used pull requests on github before, so I might've missed something - please let me know if there's any issues!

**Comment by fwSmit** [comment]
@GorrillaRibs Looks great! If you could squash the commits into the first one I'll merge the PR.

**Review by fwsmit — APPROVED** [review]

**Comment by GorrillaRibs** [comment]
Done!

**Comment by fwSmit** [comment]
Thank you :)
